src/firefly/java/edu/caltech/ipac/firefly/server/query/SearchProcessor.java

```diff
@@ -21,10 +32,63 @@
     ServerRequest inspectRequest(ServerRequest request);
     String getUniqueID(ServerRequest request);
     Type getData(ServerRequest request) throws DataAccessException;
-    void writeData(OutputStream out, ServerRequest request) throws DataAccessException;
+    FileInfo writeData(OutputStream out, ServerRequest request) throws DataAccessException;
     boolean doCache();
     void onComplete(ServerRequest request, Type results) throws DataAccessException;
     boolean doLogging();
     void prepareTableMeta(TableMeta defaults, List<DataType> columns, ServerRequest request);
     QueryDescResolver getDescResolver();
+
+//====================================================================
+//  default implementations
+//====================================================================
+
+    Logger.LoggerImpl SEARCH_LOGGER = Logger.getLogger(Logger.SEARCH_LOGGER);
+
+    static ServerRequest inspectRequestDef(ServerRequest request) {
+        TableServerRequest req = (TableServerRequest) request;
+        String doPadding = req.getMeta("padResults");
+        if (Boolean.parseBoolean(doPadding)) {
+            // if we need to pad the results, change the request.
+            req = (TableServerRequest) req.cloneRequest();
+            int start = Math.max(req.getStartIndex() - 50, 0);
+            req.setStartIndex(start);
+            req.setPageSize(req.getPageSize() + 100);
+            ((TableServerRequest)request).setStartIndex(start);   // the original request needs to be modify as well.
+            return req;
+        } else {
+            return request;
+        }
+
+    }
+
+    /**
+     * return the unique ID for the original data set of this request.  This means parameters related
+     * to paging, filtering, sorting, decimating, etc or ignored.
+     * @param request
+     * @return
+     */
+    static String getUniqueIDDef(TableServerRequest request) {
+        SortedSet<Param> params = request.getSearchParams();
+        params.add(new Param("sessID", ServerContext.getRequestOwner().getRequestAgent().getSessId()));
+        return StringUtils.toString(params, "|");
+    }
+
+    static void prepareTableMetaDef(TableMeta defaults, List<DataType> columns, ServerRequest request) {
```

Comment:
I don't think `columns` is used. May some JavaDoc here to. Is the point that there are already some defaults and this functions add more from the request? Should it be call `addMetaDefaults` or something like that.

Reply:
These *.Def in SearchProcessor are default implementations of the interface's method.  I'll go and add javadocs for them.